lib/WorkflowSmrnaseq.groovy

```diff
@@ -54,4 +54,43 @@ class WorkflowSmrnaseq {
             System.exit(1)
         }
     }
+
+    /*
+    * Format the protocol
+    * Given the protocol parameter (params.protocol),
+    * this function formats the protocol such that it is fit for the respective
+    * subworkflow
+    */
+    public static String formatProtocol(params,log) {
+
+        switch(params.protocol){
+            case 'illumina':
+                params.replace("clip_r1", 0);
```

Comment:
We should remove this and try and push it to the `modules.config` instead.

Reply:
How should this be done for a parameter that is used in many processes?

Reply:
I don't see the `clip_r1` or `three_prime_clip_r1` being used in the workflow so perhaps they can be removed.
The `MIRTRACE_RUN` process is the only one using `params.three_prime_adapter`, so I'd suggest that instead of overwrite parameters in this groovy file, we just use a switch statement inside `MIRTRACE_RUN` process to set the adapter sequence.
This would strip out a good amount of code from `WorkflowSmrnaseq.groovy`, which is always nice.

Reply:
My biggest concern with using `params.replace` is that the [replace](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#replace-K-V-) method on Java's `Map` api:
> Replaces the entry for the specified key only if it is currently mapped to some value.

... so calling `params.replace("clip_r1", 0)` will _replace_ any value set by the user with `--clip_r1 X`. 

I think what we want is the [putIfAbsent](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#putIfAbsent-K-V-) method, which:
> If the specified key is not already associated with a value (or is mapped to null) associates it with the given value and returns null, else returns the current value.

Reply:
that makes sense. and that should solve your original problem, too, right @apeltzer ? IIRC there was some issue when the parameters were null in the beginning, right?

Reply:
Yes this could work - should just test once more 👍